client/js/app.js

```diff
@@ -0,0 +1,56 @@
+const btn = document.querySelector('#post-form');
+const main = document.querySelector('main');
+
+function submitPost(e){
+    e.preventDefault();
+    const postData = {
+        title: e.target.title.value,
+        name: e.target.name.value,
+        body: e.target.body.value
+    }
+
+    const options = {
+        method: 'POST',
+        body: JSON.stringify(postData),
+        headers: {
+            "Content-Type": "application/json"
+        }
+    };
+
+    fetch('http://localhost:3000/posts/', options)
+        .then(d=>d.json())
+        .then(d => window.location.hash = `/posts/${d.id}`)
+        .catch(console.warn)
+}
+
+btn.addEventListener('submit', submitPost);
+window.addEventListener('hashchange', updateContent)
```

Comment:
Maybe could have also created an event listener for `load` and made sure that when refreshed or like using the back button of the browser the app still loads the right page. But that would also require changing the whole `updateMain` function so it could just be a time limitation element. Maybe something to consider for the future.